skyplane/compute/gcp/gcp_network.py

```diff
@@ -0,0 +1,127 @@
+from typing import List
+
+from skyplane.compute.gcp.gcp_auth import GCPAuthentication
+from skyplane.utils import imports, logger
+
+
+class GCPNetwork:
+    def __init__(self, auth: GCPAuthentication, vpc_name="skyplane"):
+        self.auth = auth
+        self.vpc_name = vpc_name
+
+    @imports.inject("googleapiclient.errors", pip_extra="gcp")
+    def create_network(errors, self):
+        compute = self.auth.get_gcp_client()
+        try:
+            compute.networks().get(project=self.auth.project_id, network=self.vpc_name).execute()
+        except errors.HttpError as e:
+            if e.resp.status == 404:  # create network
+                op = (
+                    compute.networks()
+                    .insert(project=self.auth.project_id, body={"name": self.vpc_name, "subnetMode": "auto", "autoCreateSubnetworks": True})
+                    .execute()
+                )
+                self.auth.wait_for_operation_to_complete("global", op["name"])
+            else:
+                raise e
```

Comment:
Since we are catching on `e` and potentially raising `e` again, it would be better to use the 'from None' idiom to disable automatic exception chaining:
```py
except errors.HttpError as e:
    if e.resp.status == 404:
        ...
    else:
        raise e from None
```

Reply:
@phi-line Thanks for reviewing my PR! Really excited to have you contribute to the project.

Would it be better to instead just simply call `raise` (with no argument)? What is the purpose of disabling automatic exception chaining? I'm not familiar with it.

Reply:
I'm investigating this in another branch and will add you as a reviewer for that. Thanks for this tip, it cleans up log output for Skyplane significantly.

Reply:
Fixed via #661. Thanks for the detailed insights here! It really improves the UX of the tool.